script.js

```diff
@@ -1,5 +1,202 @@
+    Click enter to start another new game`;
+    return myOutputValue;
+  }
+  else if (Number(player2diceResults.join("")) == Number(player1diceResults.join(""))){
+    myOutputValue = `Issa Draw lubba dubba dub dub <br><br>
+    Click enter to start another new game`;
+    return myOutputValue;
+  }
+};
+
+
+
+
+currentGameMode = "Welcome to Beat That"
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  console.log("test");
-  return myOutputValue;
+  if (currentGameMode == "Welcome to Beat That"){
+    myOutputValue = `Welcome to BEAT THAT!<br><br>
+     Player 1 please enter your name and click ENTER, followed by player 2 name and click ENTER to start the game. <br><br>
+      Game Rules: <br>
+      1. This is a 2 person, player vs player game. <br>
+      2. The game will roll 2 dice and show the dice rolls result. For example, dice 1 rolled 3 and dice 2 rolled 4. <br>
+      3. The player that rolled the dice will pick the order of the dice.(If they wanted the number 43, they would specify that the 2nd dice goes first.) <br>
+      4. After both players have rolled and chosen their dice order, the player with the higher combined number wins.  `;
+
+    currentGameMode = "Enter Player 1 name";
+    console.log(`currentGameMode: ${currentGameMode}`);
+
+    return myOutputValue;
+  }
+  // Take in player 1 and 2 names
+  if (currentGameMode == "Enter Player 1 name"){
+    player1 = input
+    currentGameMode = "Enter Player 2 name"
+    console.log(`Player 1: ${player1}`);
+    myOutputValue = `Welcome ${player1}. <br> Player 2, please enter your name click enter. `;
+    console.log(`currentGameMode: ${currentGameMode}`);
+    return myOutputValue
+  }
+  if (currentGameMode == "Enter Player 2 name"){
+    player2 = input
+    currentGameMode = "Player 1 turn"
+    console.log(`Player 2: ${player1}`);
+    myOutputValue = `Welcome ${player2}. <br> Player 1, press enter to start your roll `;
+    console.log(`currentGameMode: ${currentGameMode}`);
+    return myOutputValue
+  }
+  // 
+  var choiceErrorMsg = function(playerNum){
+    if(playerNum == 1){
+      return `Please enter your choice. <br><br> 
+      dice 1 : ${player1diceResults[0]} <br><br>
+      dice 2 : ${player1diceResults[1]} <br><br>
+      Please choose the order of desired dice result. <br>
+      Type 1 and click enter if you wish for dice 1 to be the first number <br>
+      Type 2 and click enter if you wish for dice 2 to be the first number`}
+    else if (playerNum == 2){
+      return `Please enter your choice. <br><br> 
+      dice 1 : ${player2diceResults[0]} <br><br>
+      dice 2 : ${player2diceResults[1]} <br><br>
+      Please choose the order of desired dice result. <br>
+      Type 1 and click enter if you wish for dice 1 to be the first number <br>
+      Type 2 and click enter if you wish for dice 2 to be the first number`
+    }
+  }
```

Comment:
again a bit repetitive, consider refactoring

the only difference is whether we are displaying from `player1diceResults` or `player2diceResults`, we prob can create variables called `dice1` and `dice2` to store the relevant value and have one single return

```
   var dice1;
   var dice2;
    if(playerNum == 1){
       dice1 = player1diceResults[0];
       dice2 = player1diceResults[1];
    } else if (playerNum == 2){
       dice1 = player2diceResults[0];
       dice2 = player2diceResults[1];
    }
    return `Please enter your choice. <br><br> 
      dice 1 : $dice1} <br><br>
      dice 2 : ${dice2} <br><br>
      Please choose the order of desired dice result. <br>
      Type 1 and click enter if you wish for dice 1 to be the first number <br>
      Type 2 and click enter if you wish for dice 2 to be the first number`